File: Tellma/ClientApp/src/app/data/entities/identity-server-user.ts
import { EntityWithKey } from './base/entity-with-key';
import { WorkspaceService } from '../workspace.service';
import { TranslateService } from '@ngx-translate/core';
import { EntityDescriptor } from './base/metadata';
import { AdminSettingsForClient } from '../dto/admin-settings-for-client';

export interface IdentityServerUser extends EntityWithKey {
    Email?: string;
    EmailConfirmed?: boolean;
    PasswordSet?: boolean;
    TwoFactorEnabled?: boolean;
    LockoutEnd?: string;
}

let _settings: AdminSettingsForClient;
let _cache: EntityDescriptor;

export function metadata_IdentityServerUser(wss: WorkspaceService, trx: TranslateService): EntityDescriptor {
    const ws = wss.admin;
    // Some global values affect the result, we check here if they have changed, otherwise we return the cached result
    if (ws.settings !== _settings) {
        _settings = ws.settings;
        _cache = {
            collection: 'IdentityServerUser',
            titleSingular: () => trx.instant('IdentityServerUser'),
            titlePlural: () => trx.instant('IdentityServerUsers'),
            select: ['Email'],
            apiEndpoint: 'identity-server-users',
            masterScreenUrl: 'identity-server-users',
            orderby: () => ['Email'],
            inactiveFilter: null,
            format: (item: IdentityServerUser) => item.Email,
            isAdmin: true,
            properties: {
                Id: { control: 'text', label: () => trx.instant('Id') },
                Email: { control: 'text', label: () => trx.instant('User_Email') },
                EmailConfirmed: { control: 'boolean', label: () => trx.instant('IdentityServerUser_EmailConfirmed') },
                PasswordSet: { control: 'boolean', label: () => trx.instant('IdentityServerUser_PasswordSet') },
                TwoFactorEnabled: { control: 'boolean', label: () => trx.instant('IdentityServerUser_TwoFactorEnabled') },
                LockoutEnd: { control: 'datetime', label: () => trx.instant('IdentityServerUser_LockoutEnd') },
            }
        };
    }

    return _cache;
}

File: Tellma.Model.Identity/IdentityServerUser.cs
using System;
using System.ComponentModel.DataAnnotations;
using Tellma.Model.Common;

namespace Tellma.Model.Identity
{
    [Display(Name = "IdentityServerUser", GroupName = "IdentityServerUsers")]
    public class IdentityServerUser : EntityWithKey<string>
    {
        [Display(Name = "User_Email")]
        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Display(Name = "IdentityServerUser_EmailConfirmed")]
        [Required]
        public bool? EmailConfirmed { get; set; }

        [Display(Name = "IdentityServerUser_PasswordSet")]
        [Required]
        public bool? PasswordSet { get; set; }

        [Display(Name = "IdentityServerUser_TwoFactorEnabled")]
        [Required]
        public bool? TwoFactorEnabled { get; set; }

        [Display(Name = "IdentityServerUser_LockoutEnd")]
        public DateTimeOffset? LockoutEnd { get; set; }
    }
}
